app/models/health.rb

```diff
@@ -0,0 +1,98 @@
+class Health
+  # things to check: delayed_job, background crons, database, ECHO, CMR, URS, Opensearch, browse-scaler
+
+  def initialize
+    @ok = true
+  end
+
+  def edsc_status
+    {ok?: @ok}
+  end
+
+  def delayed_job_status
+    # The job hasn't been run yet. Go on to check retrieval.created_at. If the time is more than 10 minutes (threshold)
+    # ago, consider the delayed job is down.
+    job = DelayedJob.last
+    retrieval = Retrieval.find_by_id(job.id)
+
+    if job.run_at.nil? && retrieval.created_at < 10.minutes.ago
```

Comment:
I see that you are focusing on the status before 10 minutes ago whereas I am looking at what happened in the past 10 minutes.

We keep all failed jobs but DelayedJob always deletes the earlier successful ones.